src/java/org/apache/cassandra/transport/Server.java

```diff
@@ -139,29 +137,27 @@ public synchronized void start()
         if (workerGroup != null)
             bootstrap = bootstrap.group(workerGroup);
 
-        if (this.useSSL)
-        {
-            final EncryptionOptions clientEnc = DatabaseDescriptor.getNativeProtocolEncryptionOptions();
+        final EncryptionOptions clientEnc = DatabaseDescriptor.getNativeProtocolEncryptionOptions();
 
-            if (clientEnc.optional)
-            {
-                logger.info("Enabling optionally encrypted CQL connections between client and server");
+        switch (this.tlsEncryptionPolicy)
+        {
+            case UNENCRYPTED:
+                bootstrap.childHandler(new Initializer(this));
+                break;
+            case OPTIONAL:
+                logger.debug("Enabling optionally encrypted CQL connections between client and server");
                 bootstrap.childHandler(new OptionalSecureInitializer(this, clientEnc));
-            }
-            else
-            {
-                logger.info("Enabling encrypted CQL connections between client and server");
+                break;
+            case ENCRYPTED:
+                logger.debug("Enabling encrypted CQL connections between client and server");
                 bootstrap.childHandler(new SecureInitializer(this, clientEnc));
-            }
-        }
-        else
-        {
-            bootstrap.childHandler(new Initializer(this));
+                break;
+            default:
+                throw new IllegalStateException("Unrecognized TLS encryption policy: " + this.tlsEncryptionPolicy);
         }
 
         // Bind and start to accept incoming connections.
-        logger.info("Using Netty Version: {}", Version.identify().entrySet());
```

Comment:
why move this log to a different location?

Reply:
It prevents it being logged twice if there is a separate SSL port specified. We only need Netty version information once.